crates/solver/src/settlement/settlement_encoder.rs

```diff
@@ -340,6 +340,24 @@ impl SettlementEncoder {
             None => U256::one().to_big_rational(),
         }
     }
+
+    /// Drops all UnwrapWethInteractions for the given token address.
+    /// This can be used in case the settlement contracts ETH buffer is big enough.
+    pub fn drop_unwrap(&mut self, token: H160) {
```

Comment:
I think there can only be one "weth" address per network. So it doesn't make sense to have the token address as a parameter here. Maybe someone else could confirm that.

Reply:
In that case wouldn't it also make sense to make `SettlementEncoder::unwraps` of type `U256` instead of `Vec<UnwrapWethInteraction`? Or maybe `UnwrapWethInteraction` if we still need the included `WETH9` struct?

Reply:
I think atm we need the weth9 instance so that we can implement the Interaction trait. With some refactoring we might be able to change that too and it probably would be slightly cleaner.

Reply:
> I think there can only be one "weth" address per network. So it doesn't make sense to have the token address as a parameter here.

The reason for this is that there are multiple "unwrap" tokens - cDAI <-> DAI, stETH <-> ETH to name a few that we potentially wanted to support here.

That being said, I would be OK with ripping this out of the the settlement encoder in favour of a single `Unwrap` interaction per settlement since we don't use this feature and it is causing confusion ATM. That being said, what I just mentioned is unrelated to this change though.

Reply:
> maybe open another PR refactoring the unwraps properly

Of course! This is what I meant with:
> what I just mentioned is unrelated to this change though